tests/tp/test_tp_strategies.py

```diff
@@ -97,8 +99,34 @@ def test_ffn_tp_strategy():
 
 
 @pytest.mark.gpu
-def test_no_tp_with_one_gpu():
-    """Test that when we have one GPU, we use DDP and not FSDP-TP."""
+@pytest.mark.world_size(4)
+@pytest.mark.parametrize('tp_strategy', ['ffn'])
+def test_tp_train(tp_strategy: str):
```

Comment:
tests like this are reasonably costly and also can be hard to debug. is there a simpler test that would achieve the same coverage?

Reply:
I believe this test is okay because
* we only train for one step
* we want to make sure things work end-to-end
* 
I spoke to @mvpatel2000 about this and he said he thought it was okay.